src/test/scala/nl/knaw/dans/easy/stage/EasyStageDatasetSpec.scala

```diff
@@ -16,156 +16,144 @@
 package nl.knaw.dans.easy.stage
 
 import java.io.File
+import java.nio.file.{Files, Paths}
 
 import nl.knaw.dans.common.lang.dataset.AccessCategory._
 import nl.knaw.dans.easy.stage.EasyStageDataset._
 import nl.knaw.dans.easy.stage.lib.Constants.DATASET_SDO
 import org.apache.commons.io.FileUtils
 import org.apache.commons.io.FileUtils.{deleteDirectory, readFileToString}
-import org.scalatest.{FlatSpec, Matchers}
+import org.scalatest.{FlatSpec, Matchers, OneInstancePerTest}
+import org.scalatest.Inside._
 
 import scala.util.{Failure, Success}
 
-class EasyStageDatasetSpec extends FlatSpec with Matchers {
+class EasyStageDatasetSpec extends FlatSpec with Matchers with OneInstancePerTest {
+  private val testDir = Paths.get("target/test", getClass.getSimpleName)
+  FileUtils.deleteQuietly(testDir.toFile)
+  Files.createDirectories(testDir)
 
-  private val tmpProps = new File("cfg/application.properties")
+  private val testBagMedium = Paths.get("src/test/resources/dataset-bags/medium")
 
   "createFileAndFolderSdos" should "stumble if the data folder does not exist" in {
+    val bagDir = testDir.resolve("bag")
+    val dataDir = bagDir.resolve("data") // Note that this directory is never created, on purpose.
+    val sdoSetDir = testDir.resolve("someSDO")
+    implicit val s = createSettings(bagDir.toFile, sdoSetDir.toFile)
+    Files.createDirectories(bagDir)
 
-    val bagitDir = new File("target/test/bag")
-    val dataDir = new File(bagitDir, "data")
-    val sdoSetDir = new File("target/test/someSDO")
-    implicit val s = createSettings(bagitDir, sdoSetDir)
-    bagitDir.mkdirs()
-
-    createFileAndFolderSdos(dataDir, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Failure[_]]
-
-    deleteDirectory(bagitDir)
+    createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Failure[_]]
   }
 
   it should "be happy with an empty data folder" in {
-
-    val bagitDir = new File("target/test/bag")
-    val dataDir = new File(bagitDir, "data")
-    val sdoSetDir = new File("target/test/someSDO")
-    implicit val s = createSettings(bagitDir, sdoSetDir)
-    dataDir.mkdirs()
-
-    createFileAndFolderSdos(dataDir, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Success[_]]
-    sdoSetDir.exists() shouldBe false
-
-    deleteDirectory(bagitDir)
+    val bagDir = testDir.resolve("bag")
+    val dataDir = bagDir.resolve("data")
+    val sdoSetDir = testDir.resolve("someSDO")
```

Comment:
Unless you have any exception to the rule, move these identical values to class level. Thus it is easier to see the differences between the tests.